packages/dds/tree/src/test/objectForest.spec.ts

```diff
@@ -95,6 +97,36 @@ function testForest(suiteName: string, factory: () => ObjectForest): void {
             assert.equal(reader.seek(1).result, TreeNavigationResult.NotFound);
         });
 
+        it("Attach basic TextCursor example to a forest", () => {
+            const forest = factory();
+            const Name = "Adam";
+            const streetName = "treeStreet";
+            const cursor = new TextCursor({
+                type: brand("Person"),
+                fields: {
+                    name: [{ value: Name, type: brand("String") }],
+                    address: [{
+                        fields: {
+                            street: [{ value: streetName, type: brand("String") }],
+                        },
+                        type: brand("Address"),
+                    }],
+                },
+            });
+            const newRange = forest.add([cursor]);
+            const dst = { index: 0, range: forest.rootField };
+            forest.attachRangeOfChildren(dst, newRange);
+
+            const reader = forest.allocateCursor();
+            reader.down("name" as FieldKey, 0);
+            assert.equal(reader.value, "Adam");
+            reader.up();
+            reader.down("address" as FieldKey, 0);
+            assert.equal(reader.value, undefined);
+            reader.down("street" as FieldKey, 0);
+            assert.equal(reader.value, streetName);
+        });
```

Comment:
I'm not including this test in my version of these changes since the bug its focused on it in textCursor and it has separate tests. For reference though I'm including below the adjustments I made to this test to make it more correct and robust:

```suggestion
    it("Attach basic TextCursor example to a forest", () => {
        const forest = factory();
        const Name = "Adam";
        const streetName = "treeStreet";
        const cursor = new TextCursor({
            type: brand("Person"),
            fields: {
                name: [{ value: Name, type: brand("String") }],
                address: [{
                    fields: {
                        street: [{ value: streetName, type: brand("String") }],
                    },
                    type: brand("Address"),
                }],
            },
        });
        const newRange = forest.add([cursor]);
        const dst = { index: 0, range: forest.rootField };
        forest.attachRangeOfChildren(dst, newRange);
    
        const reader = forest.allocateCursor();
        assert.equal(forest.tryGet(forest.root(forest.rootField), reader), TreeNavigationResult.Ok);
        assert.equal(reader.down("name" as FieldKey, 0), TreeNavigationResult.Ok);
        assert.equal(reader.value, "Adam");
        reader.up();
        assert.equal(reader.down("address" as FieldKey, 0), TreeNavigationResult.Ok);
        assert.equal(reader.value, undefined);
        assert.equal(reader.down("street" as FieldKey, 0), TreeNavigationResult.Ok);
        assert.equal(reader.value, streetName);
    });
```